ros2service/ros2service/verb/call.py

```diff
@@ -44,23 +44,31 @@ def add_arguments(self, parser, cli_name):
             help='Values to fill the service request with in YAML format ' +
                  '(e.g. "{a: 1, b: 2}"), ' +
                  'otherwise the service request will be published with default values')
-
-        def float_or_int(input_string):
-            try:
-                value = float(input_string)
-            except ValueError:
-                raise argparse.ArgumentTypeError("'%s' is not an integer or float" % input_string)
-            return value
-
         parser.add_argument(
-            '-r', '--repeat', metavar='N', type=float_or_int,
-            help='Repeat the call every N seconds')
+            '-r', '--rate', metavar='N', type=float,
+            help='Calling rate (hz) default to 1')
+        parser.add_argument(
+            '-1', '--once', action='store_true',
+            help='Cal service once and exit')
 
     def main(self, *, args):
-        return requester(args.service_type, args.service_name, args.values, args.repeat)
-
-
-def requester(service_type, service_name, values, repeat):
+        return requester(args.service_type, args.service_name, args.values, args.rate, args.once)
+
+
+def requester(service_type, service_name, values, rate, once):
+    if rate is not None:
+        if once:
+            raise RuntimeError('You cannot select both -r and -1 (--once)')
```

Comment:
This argument checking should happen within argparse. You can use a mutually exclusive group for this.